src/lib.rs

```diff
@@ -765,6 +770,63 @@ impl CertificateParams {
 						if !self.subject_alt_names.is_empty() {
 							self.write_subject_alt_names(writer.next());
 						}
+
+						// Write standard key usage
+						if !self.key_usages.is_empty() {
+							writer.next().write_sequence(|writer| {
+
+								let oid = ObjectIdentifier::from_slice(OID_KEY_USAGE);
+								writer.next().write_oid(&oid);
+								writer.next().write_bool(true);
+
+								let mut bits: u16 = 0;
+								let mut msb = 0;
+
+								for entry in self.key_usages.iter() {
+
+									// Map the index to a value
+									let index = match entry {
+										KeyUsagePurpose::DigitalSignature => 0,
+										KeyUsagePurpose::NonRepudiation => 1,
+										KeyUsagePurpose::KeyEncipherment => 2,
+										KeyUsagePurpose::DataEncipherment => 3,
+										KeyUsagePurpose::KeyAgreement => 4,
+										KeyUsagePurpose::KeyCertSign => 5,
+										KeyUsagePurpose::CrlSign => 6,
+										KeyUsagePurpose::EncipherOnly => 7,
+										KeyUsagePurpose::DecipherOnly => 8,
+									};
+
+									bits |= 1 << index;
+
+									// This helps caculate the max number of bits
+									if index > msb {
+										msb = index;
+									}
+								}
+
+								let mut nb = 1;
+								if msb > 7 {
+									nb = 2;
+								}
+
+								let mut bits = bits.to_le_bytes();
+								bits[0] = bits[0].reverse_bits();
```

Comment:
Can't this be done with `bits.to_be_bytes` instead?

Reply:
What's the reasoning for using BE over LE? Most architectures are LE, no?

Reply:
Mhh right the reverse_bits is actually needed because it is about the *bit* order and not the *byte* order. However, if you swap order of the operations it is shorter. The code `bits.reverse_bits().to_be_bytes()` is equivalent and only requires one line. It's not harder to understand it I think. I'll do it in a later change.

Reply:
Ahh yea good point. Thanks @est31